### PR TITLE
fix(op): fix payload id calculation

### PR DESCRIPTION
**Pull request by Rjected**
Attempt at fixing https://github.com/paradigmxyz/reth/issues/11729

Also updates to add the newly added eip1559 params

Right now the test from the issue fails, hopefully something simple

### CODE REVIEW

**Comment by Rjected** [comment]
EDIT: the test fails because we need to put the version as the first byte, which is sort of annoying:
```
  left: PayloadId(0x03d2dae446d2a86a)
 right: PayloadId(0x34d2dae446d2a86a)
```

from op-geth:
```go
	out[0] = byte(args.Version)
	return out
```

File: crates/optimism/payload/src/payload.rs [hunk 1]
```diff
@@ -286,16 +286,71 @@ pub(crate) fn payload_id_optimism(parent: &B256, attributes: &OpPayloadAttribute
...
-    let no_tx_pool = attributes.no_tx_pool.unwrap_or_default();
-    hasher.update([no_tx_pool as u8]);
-    if let Some(txs) = &attributes.transactions {
-        txs.iter().for_each(|tx| hasher.update(tx));
+    if attributes.no_tx_pool.unwrap_or_default() ||
+        attributes.transactions.as_ref().map(|txs| !txs.is_empty()).unwrap_or_default()
+    {
+        let no_tx_pool = attributes.no_tx_pool.unwrap_or_default();
+        hasher.update([no_tx_pool as u8]);
+        let txs_len = attributes.transactions.as_ref().map(|txs| txs.len()).unwrap_or_default();
```
Comment (mattsse) [inline crates/optimism/payload/src/payload.rs:294]: we now have multiple calls to unwrap_or_default.

can we do this just once

**Review by mattsse — APPROVED** [review]
makes sense

**Comment by avalonche** [comment]
any updates to this? seems like geth adds the payload version to the payload_id that is causing the discrepancy 

https://github.com/ethereum/go-ethereum/pull/28246

**Comment by mattsse** [comment]
unblocked now
@Rjected we can also include test cases from https://github.com/paradigmxyz/reth/pull/12089/commits/deacb4b52d9f223432d5765e11170fecaa7f2d12 that we haven't merged

**Review by mattsse — APPROVED** [review]

**Pull request merged**